### PR TITLE
feat(aapd-630): boolean with conditional. Procedure type

### PR DESCRIPTION
**Pull request by ms-ek**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-630

## Description of change

Added a boolean with a conditional text for procedure type

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/validator/validator.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const validate = () => {
 				const errors = validationResult(req);
 				const mappedErrors = errors.mapped();
 
-				if (mappedErrors.length > 0) {
```
Comment (ms-ek) [inline packages/forms-web-app/src/dynamic-forms/validator/validator.js:32]: Though it didn't cause any issues, this line wasn't actually working as far as I can see.

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/radio/question.js [hunk 2]
```diff
@@ -52,6 +52,25 @@ class RadioQuestion extends OptionsQuestion {
...
+	 * @returns {Array.<Object>}
+	 */
+	formatAnswerForSummary(sectionSegment, journey, answer) {
+		if (answer?.conditional) {
+			const selectedOption = this.options.find((option) => option.value === answer.value);
+			const conditionalAnswerText = selectedOption.conditional?.label
+				? `${selectedOption.conditional.label} ${answer.conditional}`
+				: answer.conditional;
+			const formattedAnswer = [selectedOption.text, conditionalAnswerText].join('<br>');
+			return super.formatAnswerForSummary(sectionSegment, journey, formattedAnswer, false);
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/dynamic-components/radio/question.js:70]: What's the 4th argument here?

File: packages/forms-web-app/src/dynamic-forms/validator/validator.js (hunk 1 above)
Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/validator/validator.js:32]: makes sense, undefined is never > 0 :)

**Review by Si3rr4wow — APPROVED** [review]

File: packages/forms-web-app/src/dynamic-forms/dynamic-components/radio/question.js (hunk 2 above)
Comment (ms-ek) [reply packages/forms-web-app/src/dynamic-forms/dynamic-components/radio/question.js:70]: It defines whether the answer should be capitalised or not. False means the formatted answer will remain as it is without being capitalised.

**Pull request merged**